subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java

```diff
@@ -145,6 +147,9 @@ protected void generate() {
         JavaModuleDetector javaModuleDetector = getJavaModuleDetector();
         options.classpath(new ArrayList<>(javaModuleDetector.inferClasspath(isModule, getClasspath()).getFiles()));
         options.modulePath(new ArrayList<>(javaModuleDetector.inferModulePath(isModule, getClasspath()).getFiles()));
+        if (options.getBootClasspath() != null) {
```

Comment:
Why is this still here, that should be removed, right?

Reply:
Javadoc does not use file collection. So to keep similar behavior between compile and javadoc I added warning manually also here. What do you think about that?

Reply:
How about converting the list of files to a file collection and then calling `getAsPath()` on it?

Reply:
So there is `JavadocOptionFileOption` that is used all over the place with a lot of inheritance, so I think it's not worth it doing that now. And we rather do it when we migrate to providers.

I removed this and the test. WDYT?

Reply:
I am fine with not doing anything for Javadoc. What you could do here is something like this:
```
project.files(object.getBootclasspath()).getAsPath();
```

instead of your custom validation. That would emit the deprecation warning as well.